pkg/operator/ceph/validation/validate.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	rookclient "github.com/rook/rook/pkg/client/clientset/versioned"
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"strconv"
+)
+
+var (
+	cephclusterResource   = metav1.GroupVersionResource{Group: "ceph.rook.io", Version: "v1", Resource: "cephclusters"}
+	rookClientSet         rookclient.Interface
+	universalDeserializer = serializer.NewCodecFactory(runtime.NewScheme()).UniversalDeserializer()
+)
+
+func SetRookClientSet(client rookclient.Interface) {
+	rookClientSet = client
```

Comment:
Let's avoid setting package variables except perhaps for test scenarios. In this case, you could pass the `context` with the rookClientset, or else just pass the rookClientset to the methods where needed.

Reply:
Fixed !